pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -616,6 +616,13 @@ func (p *PriorityQueue) SchedulingCycle() int64 {
 func (p *PriorityQueue) determineSchedulingHintForInFlightPod(logger klog.Logger, pInfo *framework.QueuedPodInfo) framework.QueueingHint {
 	logger.V(5).Info("Checking events for in-flight pod", "pod", klog.KObj(pInfo.Pod), "unschedulablePlugins", pInfo.UnschedulablePlugins, "inFlightEventsSize", p.inFlightEvents.Len(), "inFlightPodsSize", len(p.inFlightPods))
 
+	if len(pInfo.UnschedulablePlugins) == 0 {
```

Comment:
Check UnschedulablePlugins before checking in-flight Pods.
It's necessary because we call Done() before `AddUnschedulableIfNotPresent` and `inFlightPod` wouldn't be found.